airbyte-integrations/connectors/destination-iceberg/src/main/java/io/airbyte/integrations/destination/iceberg/config/catalog/NessieCatalogConfig.java

```diff
@@ -0,0 +1,87 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.nessie.NessieCatalog;
+import org.jetbrains.annotations.NotNull;
+
+@Data
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = false)
+public class NessieCatalogConfig
+    extends IcebergCatalogConfig {
+
+  private final String uri;
+  private final String ref;
+  private final String token;
+
+  public NessieCatalogConfig(@NotNull JsonNode catalogConfig) {
+    Preconditions.checkArgument(null != catalogConfig.get(NESSIE_CATALOG_URI_CONFIG_KEY), "%s is required", NESSIE_CATALOG_URI_CONFIG_KEY);
+    JsonNode tokenNode = catalogConfig.get(NESSIE_CATALOG_TOKEN_CONFIG_KEY);
+    JsonNode refNode = catalogConfig.get(NESSIE_CATALOG_REFERENCE_KEY);
+    this.token = null != tokenNode ? tokenNode.asText() : null;
+    this.uri = catalogConfig.get(NESSIE_CATALOG_URI_CONFIG_KEY).asText();
+    this.ref = null != refNode ? refNode.asText() : null;
+  }
+
+  @Override
+  public Map<String, String> sparkConfigMap() {
+    Map<String, String> configMap = new HashMap<>();
+    configMap.put("spark.network.timeout", "300000");
+    configMap.put("spark.sql.defaultCatalog", CATALOG_NAME);
+    configMap.put("spark.sql.extensions",
+        "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions,org.projectnessie.spark.extensions.NessieSparkSessionExtensions");
+    configMap.put("spark.sql.catalog." + CATALOG_NAME, "org.apache.iceberg.spark.SparkCatalog");
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".catalog-impl", "org.apache.iceberg.nessie.NessieCatalog");
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".uri", this.uri);
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".ref", this.ref);
+    configMap.put("spark.driver.extraJavaOptions", "-Dpackaging.type=jar -Djava.io.tmpdir=/tmp");
+
+    if (isNotBlank(this.token)) {
+      configMap.put("spark.sql.catalog." + CATALOG_NAME + ".authentication.type", "BEARER");
+      configMap.put("spark.sql.catalog." + CATALOG_NAME + ".authentication.token", this.token);
+    }
+
+    configMap.putAll(this.storageConfig.sparkConfigMap(CATALOG_NAME));
+    return configMap;
+  }
+
+  @Override
+  public Catalog genCatalog() {
+    NessieCatalog catalog = new NessieCatalog();
+    Map<String, String> properties = new HashMap<>(this.storageConfig.catalogInitializeProperties());
+    properties.put(CatalogProperties.URI, this.uri);
+    properties.put(CatalogProperties.WAREHOUSE_LOCATION, this.storageConfig.getWarehouseUri());
+    if (isNotBlank(this.ref)) {
+      properties.put("ref", this.ref);
+    }
+
+    if (isNotBlank(this.token)) {
+      properties.put(NESSIE_CATALOG_AUTHENTICATION_TYPE_KEY, "BEARER");
```

Comment:
Nessie code cannot understand `NESSIE_CATALOG_AUTHENTICATION_TYPE_KEY`. How is it working? 

It should be `CONF_NESSIE_AUTH_TYPE` from NessieConfigConstants? 

Have you verified auth with bearer token?

Reply:
+1 to reusing `NessieConfigConstants`